src/main/java/nextstep/fp/StreamStudy.java

```diff
@@ -15,19 +16,22 @@ public static long countWords() throws IOException {
                 .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
         List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
 
-        long count = 0;
-        for (String w : words) {
-            if (w.length() > 12) count++;
-        }
-        return count;
+        return words.stream().filter(w -> w.length() > 12).count();
     }
 
     public static void printLongestWordTop100() throws IOException {
         String contents = new String(Files.readAllBytes(Paths
                 .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
         List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
 
-        // TODO 이 부분에 구현한다.
+        words.stream()
+                .filter(w -> w.length() > 12)
+                .distinct()
+                .sorted(Comparator.comparing(String::length).reversed())
+                .map(String::toLowerCase)
+                .limit(100)
```

Comment:
```suggestion
                .limit(100)
                .map(String::toLowerCase)
```
이렇게 하면 어떨까요? 결과 값은 같지만 미리 100개를 제한해놓고 하면 더 좋을 것 같아요~